src/utils/api/fetch.ts

```diff
@@ -0,0 +1,77 @@
+export enum METHODS {
+  GET = "GET",
+  PUT = "PUT",
+  POST = "POST",
+  DELETE = "DELETE",
+}
+
+interface HTTPOptions {
+  timeout?: number;
+  headers?: Record<string, string>;
+  data?: Record<string, any> | null;
+}
+
+export class HTTPTransport {
+  baseUrl = "https://ya-praktikum.tech/api/v2/";
```

Comment:
```suggestion
const BASE_URL = 'https://ya-praktikum.tech/api/v2/';

export class HTTPTransport {
  private baseUrl: string;
  constructor (baseUrl: string = BASE_URL) {
      this.baseUrl = baseUrl;
  }

```
Что бы не совсем хардкод уж получался)